walker.go

```diff
@@ -0,0 +1,439 @@
+package format
+
+// TODO: This can be extracted from the `go-ipld-format` repo.
```

Comment:
This is a slightly more expectant todo than I'd personally like to have in tree :)  Maybe it can; probably it can; but we've got so many repos already that a todo suggesting yet another one gives me a bit of a willies.